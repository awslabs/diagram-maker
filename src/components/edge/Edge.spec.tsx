import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import { Edge, EdgeStyle } from '.';

describe('Edge', () => {

  it('renders the paths using props', () => {
    const edge = shallow(
      <Edge
        id="myEdge"
        edgeStyle={EdgeStyle.LEFT_RIGHT_BEZIER}
        src={{ x:0, y:100 }}
        dest={{ x:100, y:0 }}
      />
    );

    expect(edge).toMatchSnapshot();
  });

  it('passes show arrowhead', () => {
    const edge = shallow(
      <Edge
        id="myEdge"
        edgeStyle={EdgeStyle.LEFT_RIGHT_BEZIER}
        src={{ x:0, y:100 }}
        dest={{ x:100, y:0 }}
        showArrowhead={true}
      />
    );

    expect(edge).toMatchSnapshot();
  });

  it('renders selected class', () => {
    const edge = shallow(
      <Edge
        id="myEdge"
        edgeStyle={EdgeStyle.LEFT_RIGHT_BEZIER}
        src={{ x:0, y:100 }}
        dest={{ x:100, y:0 }}
        selected={true}
      />
    );

    expect(edge).toMatchSnapshot();
  });

});
