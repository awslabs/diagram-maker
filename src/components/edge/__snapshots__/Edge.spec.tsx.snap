// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edge passes show arrowhead 1`] = `
preact-render-spy (1 nodes)
-------
<g
  class="dm-edge"
  data-id="myEdge"
  data-type="DiagramMaker.Edge"
  data-event-target={true}
>
  <EdgeCurve
    src={
      Object {
        "x": 0,
        "y": 100
      }
    }
    dest={
      Object {
        "x": 100,
        "y": 0
      }
    }
    edgeStyle={0}
    class="dm-path-inner"
    showArrowhead={true}
  >
  </EdgeCurve>
  <EdgeCurve
    src={
      Object {
        "x": 0,
        "y": 100
      }
    }
    dest={
      Object {
        "x": 100,
        "y": 0
      }
    }
    edgeStyle={0}
    class="dm-path-outer"
    showArrowhead={true}
  >
  </EdgeCurve>
</g>

`;

exports[`Edge renders selected class 1`] = `
preact-render-spy (1 nodes)
-------
<g
  class="dm-edge dm-selected"
  data-id="myEdge"
  data-type="DiagramMaker.Edge"
  data-event-target={true}
>
  <EdgeCurve
    src={
      Object {
        "x": 0,
        "y": 100
      }
    }
    dest={
      Object {
        "x": 100,
        "y": 0
      }
    }
    edgeStyle={0}
    class="dm-path-inner"
  >
  </EdgeCurve>
  <EdgeCurve
    src={
      Object {
        "x": 0,
        "y": 100
      }
    }
    dest={
      Object {
        "x": 100,
        "y": 0
      }
    }
    edgeStyle={0}
    class="dm-path-outer"
  >
  </EdgeCurve>
</g>

`;

exports[`Edge renders the paths using props 1`] = `
preact-render-spy (1 nodes)
-------
<g
  class="dm-edge"
  data-id="myEdge"
  data-type="DiagramMaker.Edge"
  data-event-target={true}
>
  <EdgeCurve
    src={
      Object {
        "x": 0,
        "y": 100
      }
    }
    dest={
      Object {
        "x": 100,
        "y": 0
      }
    }
    edgeStyle={0}
    class="dm-path-inner"
  >
  </EdgeCurve>
  <EdgeCurve
    src={
      Object {
        "x": 0,
        "y": 100
      }
    }
    dest={
      Object {
        "x": 100,
        "y": 0
      }
    }
    edgeStyle={0}
    class="dm-path-outer"
  >
  </EdgeCurve>
</g>

`;
